### ISSUE TITLE
The "All Ions" control doesn't work if sim is paused

### ISSUE THREAD

**Issue by jbphet** [phetsims/neuron#100]
If the sim is paused and the "All Ions" check box is checked and unchecked, the ions don't appear or disappear.  If the user pauses the sim, turns off "All Ions", then starts stepping the sim, the background ions disappear after 10 steps.

**Comment by jbphet** [comment]
This is true regardless of whether WebGL or canvas is being used to render the particles.

**Comment by jbphet** [comment]
There is also a noticeable delay during normal operation when the "All Ions" checkbox is unchecked and the background particles disappear.

**Comment by jbphet** [comment]
I discussed this issue with @jonathanolson, and he said that it is a bit of an issue with Scenery, because Scenery is assuming that at least one thing will be drawn in the paintWebGLDrawable call and, if nothing is, no update will occur.  The scenery issue is recorded here: https://github.com/phetsims/scenery/issues/503.  We run into this because there are cases where no particles are drawn when the showAllIons flag is turned off.

**Comment by jbphet** [comment]
The trickiest test for this is to start up the sim, immediately pause it, then turn off the showing of all ions.  This is the worst case because at this point no transient particles are traveling through the leakage channels (the sim takes a little while to create the first ones).

**Comment by jbphet** [comment]
This is also a problem with the fallback node ParticlesCanvasNode, and can be tested using the ```webgl=false``` query parameter.

**Comment by jbphet** [comment]
I've addressed this issue in two major ways:
+ the nodes that draw the particles (ParticlesWebGLNode and ParticlesCanvasNode) now monitor the ```showAllIons``` property in the primary model and invalidate the paint if it changes, which forces a repaint on the next frame
+ the model now has a property that indicates whether at least one particle is present, and this property is monitored by ParticlesWebGLNode and ParticlesCanvasNode and is used to set their visibility
The second change listed above can be removed if and when https://github.com/phetsims/scenery/issues/503 is addressed.

Testing indicates that these changes address the issue.  Closing.
